contracts/seed/Seed.sol

```diff
@@ -127,132 +135,157 @@ contract Seed {
     function initialize(
         address _beneficiary,
         address _admin,
         address[] memory _tokens,
-        uint256[] memory _softAndHardCap,
+        uint256[] memory _softHardThresholds,
         uint256 _price,
         uint256 _startTime,
         uint256 _endTime,
-        uint16  _vestingDuration,
-        uint16  _vestingCliff,
+        uint32  _vestingDuration,
+        uint32  _vestingCliff,
         bool    _permissionedSeed,
         uint8   _fee
     ) public initializer {
-        beneficiary     = _beneficiary;
-        admin           = _admin;
-        softCap         = _softAndHardCap[0];
-        hardCap         = _softAndHardCap[1];
-        price           = _price;
-        startTime       = _startTime;
-        endTime         = _endTime;
-        vestingDuration = _vestingDuration;
-        vestingCliff    = _vestingCliff;
-        permissionedSeed   = _permissionedSeed;
-        seedToken       = IERC20(_tokens[0]);
-        fundingToken    = IERC20(_tokens[1]);
-        fee             = _fee;
-        closed          = false;
-        minimumReached  = false;
-        maximumReached  = false;
-        seedRemainder   = IERC20(_tokens[0]).balanceOf(address(this));
+        beneficiary       = _beneficiary;
+        admin             = _admin;
+        softCap           = _softHardThresholds[0];
+        hardCap           = _softHardThresholds[1];
+        price             = _price;
+        startTime         = _startTime;
+        endTime           = _endTime;
+        vestingDuration   = _vestingDuration;
+        vestingCliff      = _vestingCliff;
+        permissionedSeed  = _permissionedSeed;
+        seedToken         = IERC20(_tokens[0]);
+        fundingToken      = IERC20(_tokens[1]);
+        fee               = _fee;
+        closed            = false;
+        minimumReached    = false;
+        maximumReached    = false;
+
+        // Same calculation as we did at SeedFactory.deploySeed()
+        uint256 seedAmountSent = (_softHardThresholds[1].div(_price)).mul(10**18);
+        seedAmountSent = seedAmountSent.add((seedAmountSent.mul(uint256(PPM))).mul(_fee).div(PPM100));
+
+        seedRemainder     = seedAmountSent;
+        seedAmountAtStart = seedAmountSent;
     }
 
     /**
       * @dev                     Buy and lock seed tokens.
-      * @param _seedAmount       The amount of seed tokens to buy.
+      * @param _fundingAmount       The amount of seed tokens to buy.
     */
-    function buy(uint256 _seedAmount) public isActive allowedToBuy {
-        //  fundingAmount is an amount of fundingTokens required to buy _seedAmount of SeedTokens
-        uint256 fundingAmount = (_seedAmount.mul(price)).div(PCT_BASE);
+    function buy(uint256 _fundingAmount) public isActive allowedToBuy returns(uint256, uint256) {
+        // fundingAmount is an amount of fundingTokens required to buy _seedAmount of SeedTokens
+        uint256 seedAmount = (_fundingAmount.mul(PCT_BASE)).div(price);
```

Comment:
Why do we have to be doing these wei conversions? Everything here is in wei, no?